cmd/contour/serve.go

```diff
@@ -638,30 +638,26 @@ func (s *Server) doServe() error {
 	return s.mgr.Start(signals.SetupSignalHandler())
 }
 
-func (s *Server) setupTracingService(tracingConfig *contour_api_v1alpha1.TracingConfig) (*xdscache_v3.TracingConfig, error) {
-	if tracingConfig == nil {
-		return nil, nil
-	}
-
-	// ensure the specified ExtensionService exists
+func (s *Server) getExtensionSvcConfig(name string, namespace string) (xdscache_v3.ExtensionServiceConfig, error) {
 	extensionSvc := &contour_api_v1alpha1.ExtensionService{}
 	key := client.ObjectKey{
-		Namespace: tracingConfig.ExtensionService.Namespace,
-		Name:      tracingConfig.ExtensionService.Name,
+		Namespace: namespace,
+		Name:      name,
 	}
+
 	// Using GetAPIReader() here because the manager's caches won't be started yet,
 	// so reads from the manager's client (which uses the caches for reads) will fail.
 	if err := s.mgr.GetAPIReader().Get(context.Background(), key, extensionSvc); err != nil {
-		return nil, fmt.Errorf("error getting tracing extension service %s: %v", key, err)
+		return xdscache_v3.ExtensionServiceConfig{}, fmt.Errorf("error parsing extension service %s: %v", key, err)
```

Comment:
```suggestion
		return xdscache_v3.ExtensionServiceConfig{}, fmt.Errorf("error getting extension service %s: %v", key, err)
```

Reply:
Looks like you changed the wrong one; this one should be "error getting..." and the next one (for response timeout) should be "error parsing..."

Reply:
doing this...